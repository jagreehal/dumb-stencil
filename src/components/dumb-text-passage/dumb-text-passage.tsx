import { Component, Prop } from '@stencil/core';

@Component({
  tag: 'dumb-text-passage',
  styleUrl: 'dumb-text-passage.css',
  shadow: true
})
export class DumbTextPassage {
  @Prop() first: string;
  @Prop() last: string;

  render() {
    return (
      <div class="c-text-passage">
        <slot/>
      </div>
    );
  }
}
