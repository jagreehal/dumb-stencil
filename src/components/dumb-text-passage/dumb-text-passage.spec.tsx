import { TestWindow } from '@stencil/core/testing';
import { DumbTextPassage } from './dumb-text-passage';

describe('dumb-text-passage', () => {
  let element;
  let window;

  beforeEach(async () => {
    window = new TestWindow();
    element = await window.load({
      components: [DumbTextPassage],
      html:  `
      <dumb-text-passage>
        <p>A text passage contains arbitrary text that might come from a CMS. It should live within a container that caps the line length of the text to avoid a straining reading experience.</p>      
      </dumb-text-passage>`
    });
  });

  it('should match snapshot', async () => {
    expect(element).toMatchSnapshot();
  });
});
