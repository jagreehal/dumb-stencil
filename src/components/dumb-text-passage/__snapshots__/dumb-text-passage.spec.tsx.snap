// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dumb-text-passage should match snapshot 1`] = `
<dumb-text-passage
  class="hydrated"
>
  
  
  
  
  <div
    class="c-text-passage"
  >
    
    
        
    <p>
      A text passage contains arbitrary text that might come from a CMS. It should live within a container that caps the line length of the text to avoid a straining reading experience.
    </p>
          
      
  </div>
</dumb-text-passage>
`;
